File: src/main/resources/static/app/app/ProductController.js
///* 
// * To change this license header, choose License Headers in Project Properties.
// * To change this template file, choose Tools | Templates
// * and open the template in the editor.
// */
//
//var app = angular.module('ProductController',[]).controller('ProductController',function ($http,$scope,$location,$rootScope){
//   
//    
//    var productList = [];
//   
//    alert("dddddddddddd");
//    $scope.init = function(){
//     $scope.listProduct(); 
//    };
//    
//    $scope.listProduct = function(){
//        var url = "admin/listproduct";
//         alert("dddddddddddd   list");
//        var productProcess = $http.get(url);
//        productProcess.then(function(response){
//        productList = response.data;; 
//        });   
//    };
//    
//     $scope.addProduct = function(){
//            var url = "admin/addproduct";
//            var product = $scope.pro;
//             
//             var productProcess = $http.post(url,product).then(function(response){
//                 alert("dddddddddddd");  
//                alert("product has been added"); 
//                //$rootScope.status = response.data.status;
//                $rootScope.pname = response.data.item.productName;
//               // productList = response.data.item;
//                alert($rootScope.pname);
//               
//            });
//            
//             
//        }; 
//      $scope.deleteProduct = function(productId){
//        var url = "Admin/deleteProduct" + productId;
//        
//        $http.post(url).then(function (response){
//            
//            v.product = response.data;
//        });
//    };    
//              
//    
//});

File: src/main/java/com/mrd/controller/ProductController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mrd.controller;

import com.mrd.entity.Product;
import com.mrd.entity.User;
import com.mrd.service.ProductService;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PathVariable;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController

public class ProductController {
   
    @Autowired
    private ProductService prodService;

    List<Product> prodList = new ArrayList<>();
    /*Fecthing all products*/
    @RequestMapping(value="/products/all",method = RequestMethod.GET)
    public List<Product> allProducts(){
      return prodService.listProducts();
    }
    
    /*saving product */
    @RequestMapping(value = "/addProduct",method = RequestMethod.POST)
    public void addProduct(@RequestBody Product product){
        prodService.addProduct(product);
    }

  /*get product by user_id*/
    @RequestMapping(value="/rest/{user_id}/product", produces ={MediaType.APPLICATION_JSON_VALUE}) 
   public Set<Product> byUser(@PathVariable int user_id){
    return prodService.listProduct(user_id);
}
   

}
    

